NumberTheory/GCD.cpp

```diff
@@ -0,0 +1,19 @@
+// C program to demonstrate Basic Euclidean Algorithm
```

Comment:
It's in **C++**, right?! :smile: 

Also, it would be better to write it in the format described in issue #8, mentioning the time and space complexities.